File: benchmark/gb.js
const rpn = require('request-promise-native')
async function fetchBooks (genres) {
  for (let i = 0; i < genres.length; i++) {
    const genre = genres[i]
    const body = await rpn('http://localhost:3000/books?genreId=' + genre.id)
    const books = JSON.parse(body).map((book) => book.title)
    genres[i] = {name: genre.name, books}
  }
  console.log(JSON.stringify(genres))
}
async function fetchGenres () {
  const body = await rpn('http://localhost:3000/genres')
  const genres = JSON.parse(body)
  fetchBooks(genres)
}
fetchGenres()

File: benchmark/gb.py
import json
from urllib import request
response = request.urlopen('http://localhost:3000/genres').read()
genres = json.loads(response)
result = []
for genre in genres:
    genre_row = {'genre': genre['name'], 'books': []}
    response = request.urlopen('http://localhost:3000/books?genreId=' + str(genre['id'])).read()
    books = json.loads(response)
    genre_row['books'] = list(map(lambda x: x['title'], books))
    result.append(genre_row)
print(json.dumps(result))
